### PR TITLE
Round date range date times

### PR DESCRIPTION
**Pull request by kouknick**
## Description
Fix inconsistent DateRangeInput results, by rounding the start and end date times.
- Start time is rounded to `00.000` 
- End time is rounded to `59.999`
- If time picker is not enabled start and end times are rounded to the start and the end of the day

Closes #1921

### Checklist:

- [x] Make sure you give an high-level overview of the problem & the solution
- [x] Make sure the related issue is referenced
- [x] Make sure screenshots are added if the UI is changed in any way
- [x] Make sure all of the significant logic is covered by tests
- [x] Make sure your changes are accessible (addition of aria-attributes & correct HTML semantics)
- [x] Make sure you reference the related design document if it exists
- [ ] Make sure telemetry is added (analytics & errors)

### CODE REVIEW

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 1]
```diff
@@ -248,6 +248,72 @@ it.only('allows changing time options in 24h mode', async () => {
   expect(ending.format('DD/MM/YYYY HH:mm A')).toBe('11/11/2020 00:22 AM');
 });
 
+it.only('rounds start and end dates by minute', async () => {
```
Comment (vorillaz) [inline src/components/DateRangeInput/DateRangeInput.test.tsx:251]: You forgot an `.only` here :D

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 2]
```diff
@@ -248,6 +248,72 @@ it.only('allows changing time options in 24h mode', async () => {
...
+
+  const submitBtn = await findByText('Apply');
+  await fireEvent.click(submitBtn);
+  expect(mock).toHaveBeenCalledWith([
+    start.startOf('minute').toDate(),
+    endDate.hour(12).minute(30).endOf('minute').toDate(),
+  ]);
+});
+
+it.only('rounds start and end dates by day', async () => {
```
Comment (vorillaz) [inline src/components/DateRangeInput/DateRangeInput.test.tsx:289]: Ditto.

File: src/components/DateRangeInput/DateRangeInput.tsx [hunk 3]
```diff
@@ -126,9 +126,16 @@ const DateRangeInput: React.FC<
 
   const onApply = useCallback(
     e => {
+      // To avoid inconsistent results round the start and end dates to the
+      // nearest minute (or day if the time picker is disabled)
+      const timeUnit = withTime ? 'minute' : 'day';
+      const startEndDates = [
+        dayjs(currentDateRange[0]).startOf(timeUnit).toDate(),
+        dayjs(currentDateRange[1]).endOf(timeUnit).toDate(),
```
Comment (vorillaz) [inline src/components/DateRangeInput/DateRangeInput.tsx:134]: I am not pretty sure if the `currentDateRange array has always two items, can we confirm this one?

**Review by vorillaz — COMMENTED** [review]
Overall looking good, with some minor notes here and there. I loved the testing part. Great job!

File: src/components/DateRangeInput/DateRangeInput.tsx (hunk 3 above)
Comment (kouknick) [reply src/components/DateRangeInput/DateRangeInput.tsx:134]: The Apply button is disabled until the users fill in both dates, is this enough or should I add an extra validation in the `onApply` function?

```
 const isDisabled = useCallback(() => !currentDateRange || currentDateRange?.length < 2, [
    currentDateRange,
  ]);
```
               
```
 <Button disabled={isDisabled()} onClick={onApply} size="medium">
       Apply
</Button>
```

File: src/components/DateRangeInput/DateRangeInput.test.tsx (hunk 1 above)
Comment (kouknick) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:251]: I added `.only` because most of the tests in that file are skipped.

**Comment by 3nvi** [comment]
@kouknick @vorillaz we have to fix this asap. Please prioritize it

**Review by vorillaz — APPROVED** [review]

**Comment by rleighton** [comment]
Please have the indicator search re tested for the release

**Pull request merged**